toucan_connectors/net_explorer/net_explorer_connector.py

```diff
@@ -0,0 +1,118 @@
+import json
+import logging
+from io import BytesIO
+from typing import Optional
+
+import pandas as pd
+import requests
+from pydantic import Field, SecretStr
+
+from toucan_connectors.common import ConnectorStatus
+from toucan_connectors.toucan_connector import ToucanConnector, ToucanDataSource
+
+
+class NetExplorerDataSource(ToucanDataSource):
+    file: str
+    sheet: Optional[str] = 0
+
+
+class NetExplorerConnector(ToucanConnector):
+    data_source_model: NetExplorerDataSource
+    instance_url: str = Field(
+        None,
+        Title='Instance URL',
+        placeholder='exemple.netexplorer.pro',
+    )
+    user: str
+    password: SecretStr
+
+    def _retrieve_token(self):
+        login_url = f'https://{self.instance_url}/api/auth'
+
+        data = json.dumps({'user': self.user, 'password': self.password.get_secret_value()})
+
+        headers = {'Content-Type': 'application/json'}
+
+        resp = requests.request('POST', login_url, data=data, headers=headers)
+
+        return resp.json()['token']
+
+    def _retrieve_folders(self, token):
+        folders_url = f'https://{self.instance_url}/api/folders?depth=-1'
+
+        headers = {'Authorization': f'Bearer {token}'}
+
+        resp = requests.request('GET', folders_url, data={}, headers=headers)
+
+        return resp.json()
+
+    def _retrieve_file_id(self, folders, data_source):
+
+        basedir = data_source.file.split('/')[0]
+        path = data_source.file.split('/')[1:]
+
+        id = None
```

Comment:
Because `id` is a python reserved keyword
```suggestion
        _id = None
```